core/update.php

```diff
@@ -342,6 +316,116 @@ function update_info_page() {
   return $output;
 }
 
+/**
+ * Provides a form to create an on-demand backup before updating.
+ *
+ * @return string
+ *   Rendered HTML form.
+ */
+function update_backup_page() {
+  update_task_list('backup');
+  backdrop_set_title('Pre-update backup');
+
+  $elements = backdrop_get_form('update_backup_form');
+  return backdrop_render($elements);
+}
+
+
+/**
+ * Form constructor for the list of available database module updates.
+ */
+function update_backup_form($form, &$form_state) {
+  // Check if Backup directory is specified. If FALSE, this form shouldn't even
+  // be displayed. If NULL, link to documentation on setting up backups.
+  $backup_directory = settings_get('backup_directory');
+
+  $help = '<p>' . t('Before running updates, it is recommended to create a backup of your database and configuration.') . '</p>';
+  $help .= '<p>' . t('If skipping the backup process, please ensure you create a backup through a different mechanism, such as through your hosting provider.') . '</p>';
+
+  if (empty($backup_directory)) {
+    $help .= '<p>' . t('Backups are not available because the variable !variable has not been set in !file.', array(
+      '!variable' => '<code>$settings[\'backup_directory\']</code>',
+      '!file' => '<code>settings.php</code>',
+    )) . '</p>';
+    $help .= '<p>' . t('Please check the <a href="!url">documentation on configuring backups</a>.', array(
+      '!url' => 'https://docs.backdropcms.org/documentation/backup-and-restore',
+    )) . '</p>';
+  }
+  else {
+    $help .= '<p>' . t('The backup process may take several minutes, depending on the size of your database.') . '</p>';
+  }
+  $form['help'] = array(
+    '#type' => 'help',
+    '#markup' => $help,
+    '#weight' => -5,
+  );
+
+  $backup_targets = settings_get('backup_targets', array());
+  $backup_targets = array_merge($backup_targets, array(
+    'db:default',
+    'config:active',
+  ));
+
+  $form['backups'] = array(
+    '#tree' => TRUE,
+  );
+
+  foreach ($backup_targets as $backup_target) {
+    list($backup_plugin_id, $backup_subkey) = explode(':', $backup_target, 2);
+    $backup_class = backup_get_handler_name($backup_target);
+    if (!$backup_class) {
+      continue;
+    }
+
+    /** @var Backup $backup_instance */
```

Comment:
No var $backup_instance here, as far as I can see no instance of Backup, either. Copy-paste zombie?

Reply:
Thanks, code refactor issue where `backup_get_handler_name()` once was `backup_get_handler()` and returned a class.